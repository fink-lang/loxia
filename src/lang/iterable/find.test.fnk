{fink2js} = import '../../testing'
{describe, it, expect, to_match_snapshot} = import '@fink/jest'


describe 'find', fn:
  it 'compiles simple find', fn:
    expect
      fink2js`
        find item:
          item > 3
      `
      to_match_snapshot


  it 'compiles complex find', fn:
    expect
      fink2js`
        find {item}:
          match item:
            {spam}: shrub
            else: ni

        find [foo, ...bar]:
          foo_alone = foo && bar.length == 0
          foo_alone

        find [...foo, bar]:
          bar && foo.length == 0
      `
      to_match_snapshot




