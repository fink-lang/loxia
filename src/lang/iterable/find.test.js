import {fink2js} from '../../testing';


describe('compiles fold', ()=> {
  it('compiles simple find', ()=> {
    expect(
      fink2js(`
        find item:
          item > 3
      `)
    ).toMatchSnapshot();
  });

  it('compiles complex find', ()=> {
    expect(
      fink2js(`
        find {item}:
          match item:
            {spam}: shrub
            else: ni
      `)
    ).toMatchSnapshot();
  });
});


