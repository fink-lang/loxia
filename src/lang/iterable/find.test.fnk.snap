// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find compiles complex find 1`] = `
"ˆitems_2 => {
  for (const ˆitem_1 of ˆitems_2) {
    const {
      \\"item\\": item
    } = ˆitem_1;

    let _do_result;

    ˆmatch_5: {
      const ˆvalue_4 = item;

      if (ˆvalue_4 !== undefined && ˆvalue_4 !== null) {
        const {
          \\"spam\\": ˆp_6
        } = ˆvalue_4;

        if (ˆp_6 === spam) {
          _do_result = shrub;
          break ˆmatch_5;
        }
      }

      {
        _do_result = ni;
        break ˆmatch_5;
      }
    }

    const ˆfound_3 = _do_result;
    _do_result = undefined;
    if (ˆfound_3) return ˆitem_1;
  }

  return undefined;
};

ˆitems_8 => {
  for (const ˆitem_7 of ˆitems_8) {
    const [foo, ...bar] = ˆitem_7;
    const foo_alone = foo && bar.length === 0;
    const ˆfound_9 = foo_alone;
    if (ˆfound_9) return ˆitem_7;
  }

  return undefined;
};

ˆitems_12 => {
  for (const ˆitem_10 of ˆitems_12) {
    let _do_result2;

    {
      const [...ˆitems_11] = ˆitem_10;
      _do_result2 = [ˆitems_11.slice(0, -1), ˆitems_11.slice(-1)];
    }
    const [foo, [bar]] = _do_result2;
    _do_result2 = undefined;
    const ˆfound_13 = bar && foo.length === 0;
    if (ˆfound_13) return ˆitem_10;
  }

  return undefined;
};

Object.assign(module.exports, {});"
`;

exports[`find compiles simple find 1`] = `
"ˆitems_2 => {
  for (const ˆitem_1 of ˆitems_2) {
    const item = ˆitem_1;
    const ˆfound_3 = item > 3;
    if (ˆfound_3) return ˆitem_1;
  }

  return undefined;
};

Object.assign(module.exports, {});"
`;
